modules/custom/az_finder/src/Form/AZFinderSettingsForm.php

```diff
@@ -0,0 +1,412 @@
+        '#type' => 'submit',
+        '#value' => $this->t('Delete'),
+        '#ajax' => [
+          'callback' => '::ajaxDeleteOverride',
+          'wrapper' => 'js-overrides-container',
+          'effect' => 'fade',
+        ],
+        '#name' => 'delete-' . $key,
+        '#submit' => ['::submitDeleteOverride'],
+        '#attributes' => [
+          'class' => ['button--small'],
+        ],
+      ];
+
+      $config_name = "az_finder.tid_widget.{$view_id}.{$display_id}";
+      $config = $this->config($config_name);
+      $vocabulary_ids = $this->azFinderVocabulary->getVocabularyIdsForFilter($view_id, $display_id, 'taxonomy_index_tid');
+
+      foreach ($vocabulary_ids as $vocabulary_id) {
+        $this->azFinderVocabulary->addTermsTable(
+              $form['az_finder_tid_widget']['overrides'][$key]['vocabularies'][$vocabulary_id],
+              $vocabulary_id,
+              $view_id,
+              $display_id
+          );
+
+      }
+
+    }
+    $overrides = $form_state->getValue(['az_finder_tid_widget', 'overrides']) ?? [];
+    $form_state->setValue(['az_finder_tid_widget', 'overrides'], $overrides);
+    return $form['az_finder_tid_widget']['overrides'][$key];
+  }
+
+  /**
+   * Ajax callback for the delete button.
+   */
+  public function ajaxDeleteOverride(array &$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    $button_name = $triggering_element['#name'];
+    $key = str_replace('delete-', '', $button_name);
+    $overrides = $form_state->getValue(['az_finder_tid_widget', 'overrides']) ?? [];
+    unset($overrides[$key]);
+    unset($form['az_finder_tid_widget']['overrides'][$key]);
+    $this->configFactory->getEditable('az_finder.tid_widget.' . $key)->delete();
+    $form_state->setValue(['az_finder_tid_widget', 'overrides'], $overrides);
+
+    // Rebuild the form.
+    $form_state->setRebuild(TRUE);
+
+    return $form['az_finder_tid_widget']['overrides'];
+  }
+
+  /**
+   * Separate submit handler for the delete button.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   */
+  public function submitDeleteOverride(array &$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    $button_name = $triggering_element['#name'];
+    $key = str_replace('delete-', '', $button_name);
+    [$view_id, $display_id] = explode(':', $key);
+    $config_name = "az_finder.tid_widget.$view_id.$display_id";
+    $config = $this->config($config_name);
+    if ($config) {
+      $editable_config = $this->configFactory->getEditable($config_name);
+      $editable_config->delete();
+    }
+    // Update the overrides in form state.
+    $overrides = $form_state->getValue(['az_finder_tid_widget', 'overrides']) ?? [];
+    unset($overrides[$key]);
+    $form_state->setValue(['az_finder_tid_widget', 'overrides'], $overrides);
```

Comment:
```suggestion
    $form_state->setValue(['az_finder_tid_widget', 'overrides'], $overrides);
    $form_state->setRebuild(TRUE);
```

Reply:
I'll move these commit suggestions to a separate PR based on this branch to make it easier to review them.